staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -122,6 +123,18 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope *RequestSc
 		}
 		options.TypeMeta.SetGroupVersionKind(metav1.SchemeGroupVersion.WithKind("DeleteOptions"))
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.AllowUnsafeMalformedObjectDeletion) {
+			if options != nil && ptr.Deref(options.IgnoreStoreReadErrorWithClusterBreakingPotential, false) {
+				p, ok := r.(rest.CorruptObjectDeleterProvider)
+				if !ok {
+					// TODO: this is a developer error
+					scope.err(fmt.Errorf("ignoreStoreReadErrorWithClusterBreakingPotential is not supported"), w, req)
+					return
+				}
+				r = p.GetCorruptObjDeleter()
```

Comment:
@sttts should we say `internal error, ignoreStoreReadErrorWithClusterBreakingPotential is not initialized"` instead?

Also, do we want to add?
```
deleter := p.GetCorruptObjDeleter()
if deleter == nil {
     scope.err(fmt.Errorf("internal error, ignoreStoreReadErrorWithClusterBreakingPotential is not initialized"), w, req)
     return
}
```

Reply:
hot take: should we add it to the graceful deleter interface?

Reply:
Only downsides: ppl might not notice that they don't support the new GracefulDeleter interface anymore.

Reply:
In theory you could check before the closure that `r` has the interface. That non-nil is return by the methods you can specify in the interface.

Reply:
fixed it